README.md

```diff
@@ -811,6 +813,23 @@ RuboCop::RakeTask.new(:rubocop) do |task|
 end
 ```
 
+## Caching
+
+Large projects containing hundreds or even thousands of files can take
+a really long time to inspect, but RuboCop has functionality to
+mitigate this problem. A result caching mechanism is activated if the
+configuration parameter `AllCops: UseCache` is `true`, which it is by
+default. The command line option `--cache false` can be used to turn
+off result caching, thus overriding the configuration parameter. If
+`AllCops: UseCache` is set to `false` in the local `.rubocop.yml`,
+then it's `--cache true` that overrides the setting.
+
+The cache is stored in `.rubocop_cache/` under the current directory.
+There are subdirectories for each user, so as to minimize the risk of
+collisions and file curruption from simultaneous `rubocop`
+invocations. Git users should add `.rubocop_cache` to their
```

Comment:
@bbatsov Yes, I will try to implement something along those lines. I like @kbrock's idea about renaming files as a way to make the operation atomic. I think it could work, and I'm going to try it.